test-cases/python/web-security-issues.py

```
@@ -0,0 +1,152 @@
+    return len(password) >= 4
+
+def validate_api_key(api_key):
+    return api_key == "SECRET_API_KEY_123"
+
+def create_user(user_data):
+    user = {
+        'email': user_data['email'],
+        'password': user_data['password'],
+        'role': user_data['role']
+    }
+    save_user(user)
```

Ruff (0.12.2)

Undefined name `save_user`

(F821)